packages/nc-gui/composables/useUIPermission/rolePermissions.ts

```diff
@@ -1,51 +1,65 @@
 import { ProjectRole, Role } from '~/lib'
 
-const rolePermissions = {
+const rolePermissions: Record<Role | ProjectRole, '*' | Partial<Record<'include' | 'exclude', Record<string, boolean>>>> = {
   // general role permissions
   /** todo: enable wildcard permission
    *  limited permission  due to unexpected behaviour in shared base if opened in same window  */
-  [Role.Super]: {
-    projectTheme: true,
-  },
+  [Role.Super]: '*',
   [Role.Admin]: {},
   [Role.Guest]: {},
   [Role.User]: {
-    projectCreate: true,
-    projectActions: true,
-    projectSettings: true,
+    include: {
+      projectCreate: true,
+      projectActions: true,
+      projectSettings: true,
+    },
   },
 
   // Project role permissions
-  [ProjectRole.Creator]: '*',
-  [ProjectRole.Owner]: '*',
+  [ProjectRole.Creator]: {
+    exclude: {
+      appStore: true,
+    },
+  },
+  [ProjectRole.Owner]: {
+    exclude: {
+      appStore: true,
+    },
+  },
   [ProjectRole.Editor]: {
-    smartSheet: true,
-    xcDatatableEditable: true,
-    column: true,
-    tableAttachment: true,
-    tableRowUpdate: true,
-    dataInsert: true,
-    rowComments: true,
-    gridViewOptions: true,
-    sortSync: true,
-    fieldsSync: true,
-    gridColUpdate: true,
-    filterSync: true,
-    csvImport: true,
-    apiDocs: true,
-    projectSettings: true,
-    newUser: false,
+    include: {
+      smartSheet: true,
+      xcDatatableEditable: true,
+      column: true,
+      tableAttachment: true,
+      tableRowUpdate: true,
+      dataInsert: true,
+      rowComments: true,
+      gridViewOptions: true,
+      sortSync: true,
+      fieldsSync: true,
+      gridColUpdate: true,
+      filterSync: true,
+      csvImport: true,
+      apiDocs: true,
+      projectSettings: true,
+      newUser: false,
+    },
   },
   [ProjectRole.Commenter]: {
-    smartSheet: true,
-    column: true,
-    rowComments: true,
-    projectSettings: true,
+    include: {
+      smartSheet: true,
+      column: true,
+      rowComments: true,
+      projectSettings: true,
+    },
   },
   [ProjectRole.Viewer]: {
-    smartSheet: true,
-    column: true,
-    projectSettings: true,
+    include: {
+      smartSheet: true,
+      column: true,
+      projectSettings: true,
+    },
   },
 } as const
 
```

Comment:
Update the below types to so the typing works properly again, otherwise `Permission` will be resolved to `never` with your changes.

```ts
type RolePermissions = Omit<typeof rolePermissions, 'guest' | 'admin' | 'super'>

type GetKeys<T> = T extends Record<any, Record<infer Key, boolean>> ? Key : never

export type Permission<K extends keyof RolePermissions = keyof RolePermissions> = RolePermissions[K] extends Record<any, any>
  ? GetKeys<RolePermissions[K]>
  : never

export default rolePermissions
```